feat(clickpipes): document custom private DNS for AWS PrivateLink reverse private endpoints (#6492)
ClickHouse/clickhouse-docs: docs/integrations/data-ingestion/clickpipes/aws-privatelink.md

can be configured for ClickPipes. Add [your ClickPipe region](#aws-privatelink-regions) to the allowed regions in your VPC endpoint service.
:::

## Custom private DNS {#custom-private-dns}

<PrivatePreviewBadge/>

:::info

> Custom private DNS is in private preview. Contact the ClickHouse support team to enable it for your service.

Reviewer: ```suggestion
Custom private DNS is in **Private Preview**. Reach out to the [ClickHouse support team](https://clickhouse.com/support/program) to enable it for your service.
```

:::

AWS PrivateLink provides [managed private DNS](https://docs.aws.amazon.com/vpc/latest/privatelink/manage-dns-names.html)
for VPC endpoint services with a verified domain. Some services can't use it and instead
require each consumer to manage DNS resolution in their own VPC — for example, Confluent Cloud

> requires consumers to resolve broker hostnames to the PrivateLink endpoint themselves.

Reviewer: ```suggestion
AWS PrivateLink provides [managed private DNS](https://docs.aws.amazon.com/vpc/latest/privatelink/manage-dns-names.html)
for VPC endpoint services with a verified domain. Some services require each consumer to manage DNS resolution in their own VPC, e.g., Confluent Cloud requires consumers to resolve broker hostnames to the PrivateLink endpoint.
```

For these cases, ClickPipes supports attaching custom private DNS names to a reverse private
endpoint. ClickPipes resolves these names to the endpoint's private addresses, so your data
source can be reached by its own hostname over private connectivity.

Custom private DNS complements managed private DNS — it doesn't replace it. If your
PrivateLink service already provides private DNS names, you don't need custom names.

Custom private DNS names are supported for the [VPC endpoint service](#vpc-endpoint-service)
and [VPC resource](#vpc-resource) endpoint types. MSK multi-VPC provides managed private DNS

> out of the box and doesn't support custom names.

Reviewer: ```suggestion
out-of-the-box and doesn't support custom names.
```

The following rules apply to custom private DNS names:

- Exact names (`kafka.internal.example.com`) and wildcard names (`*.example.com`) are
  supported. A wildcard matches a single DNS label — for example,
  `*.abcde12345.us-east-1.aws.confluent.cloud` matches
  `b0-lkc123.abcde12345.us-east-1.aws.confluent.cloud`.
- Names must be unique across all reverse private endpoints of a ClickHouse service,
  including overlaps between wildcard and exact names.
- Names under reserved suffixes (such as `local`, `localhost`, `internal`, `corp`,

>   `private`) are rejected.

Reviewer: ```suggestion
- Names under reserved suffixes (e.g., `local`, `localhost`, `internal`, `corp`,
  `private`) are rejected.
```

    - For VPC resource, provide the configuration share ARN and configuration ID.
    - For MSK multi-VPC, provide the cluster ARN and authentication method used with a created endpoint.
    - For VPC endpoint service, provide the service name.

>     - Optionally, provide a custom private DNS name — see [Custom private DNS](#custom-private-dns).

Reviewer: ```suggestion
    - Optionally, provide a [custom private DNS](#custom-private-dns) name.
```

8. Once the endpoint is ready, you can use a DNS name to connect to the data source.

   On a list of endpoints, you can see the DNS name for the available endpoint.
   It can be an internally ClickPipes provisioned DNS name, a private DNS name supplied by a PrivateLink service,

>    or a [custom private DNS name](#custom-private-dns) you configured.

Reviewer: ```suggestion
   or a [custom private DNS name](#custom-private-dns).
```
